File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/MappersModule.java
package ru.mamykin.foreignbooksreader.di.modules;

import dagger.Module;
import dagger.Provides;
import ru.mamykin.foreignbooksreader.common.FileToAndroidFileMapper;
import ru.mamykin.foreignbooksreader.common.FolderToFilesListMapper;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
public class MappersModule {
    @Provides
    FolderToFilesListMapper provideFolderToFilesListMapper() {
        return new FolderToFilesListMapper();
    }

    @Provides
    FileToAndroidFileMapper provideFileToAndroidFileMapper() {
        return new FileToAndroidFileMapper();
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/MappersModule.kt
package ru.mamykin.foreignbooksreader.di.modules

import dagger.Module
import dagger.Provides
import ru.mamykin.foreignbooksreader.common.FileToAndroidFileMapper
import ru.mamykin.foreignbooksreader.common.FolderToFilesListMapper

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
class MappersModule {
    @Provides
    internal fun provideFolderToFilesListMapper(): FolderToFilesListMapper {
        return FolderToFilesListMapper()
    }

    @Provides
    internal fun provideFileToAndroidFileMapper(): FileToAndroidFileMapper {
        return FileToAndroidFileMapper()
    }
}